New tree refactor

jvigliotta opened the pull request:
This new tree addresses the performance issues apparent when viewing many items (children, nodes) at one time as well as make the navigation tree more clean. It does this by showing one directory (and its ancestors at a time.

EDIT: 
Moved search functionality to the same tree so it will also utilize virtual scroll feature.

Address issue #3060 

Changes address original issue? Yes
Unit tests included and/or updated with changes? Coming Soon
Command line build passes? Yes
Changes have been smoke-tested? Yes
Testing instructions included? Yes (on issue)

deeptailor commented on src/ui/layout/mct-tree.vue:
```diff
@@ -57,43 +99,275 @@
+            }
+        },
+        setContainerHeight() {
+            if(this.$refs.mainTree) {
```
Using vue.$nextTick should be a more predictable way to know when the tree is rendered

jvigliotta replied:
Yep, changed to nextTick and that covered the replacement!

jvigliotta commented on src/ui/layout/BrowseBar.vue:
```diff
@@ -271,6 +275,9 @@ export default {
             window.location.hash = this.parentUrl;
+        },
+        syncNavigationTree() {
+            this.$emit('syncTreeNavigation');
```
Done!

deeptailor commented:
@charlesh88 - this PR looks ready enough to get your styles blessing

charlesh88 commented:
@jvigliotta @deeptailor I'm going to go in and mess with the Sync button, will push an update soon.

jvigliotta commented:
@charlesh88 Update: Removed arrows on search results, fixed "My Items" not highlighting when navigated to, added an "Empty" notice when a folder doesn't have children. This will need your styling skillz...

charlesh88 commented:
I have also made a lot of changes in this branch:

### Changes
- Significant consolidation and cleanups in mct-tree.scss;
- Normalize base and hover styles across new tree, legacy tree, list-items (used in Notebook) and Folder List View;
- Add styling to override and remove `<a> outline: dotted` coming from normalize-min;  remove too-broad `<a>` hover styling from global.scss;
- Fix styles for Snow theme; updated and synced Maelstrom and Espresso themes;
- Disallow pointer-events on `is-navigated` object's label (click on `c-nav__down element` still allowed);
- Simplified and improved markup and CSS for pane collapse/expand buttons;
- Added new named slot to pane.vue; moved tree sync button into tree pane, new icon and look-and-feel for pane collapse buttons;
- Normalized Status Area expand/collapse look-and-feel to match new approach in panes, added `icon-items-collapse` and `icon-items-expand` for parity with design changes;
- Styled 'empty' tree item element;

### Author Checklist
1. Changes address original issue? Y
1. Unit tests included and/or updated with changes? N/A
1. Command line build passes? Y
1. Changes have been smoke-tested? Y

deeptailor commented on src/ui/layout/mct-tree.vue:
```diff
@@ -10,107 +13,452 @@
+            <!-- end loading -->
+        </div>
+        <!-- currently viewed children -->
+        <transition
```
This is cool

jvigliotta replied:
Yeah I was stoked when I found that. Really came through here.

deeptailor commented on src/ui/layout/mct-tree.vue:
```diff
@@ -10,107 +13,452 @@
+                            :item-index="index"
+                            :item-height="itemHeight"
+                            :virtual-scroll="!noScroll"
+                            :show-down="activeSearch ? false : true"
```
we can turn this into a computed value, for better readability

deeptailor commented on src/ui/layout/mct-tree.vue:
```diff
@@ -10,107 +13,452 @@
+                        />
+                        <li
+                            v-if="visibleItems.length === 0"
+                            :style="{ paddingLeft: ((ancestors.length + 1) * 10) + 'px' }"
```
can we create a method that does this? with a descriptive method name, it will be easy to read

jvigliotta replied:
Apparently the function existed already "emptyStyles()" so hooked that up!

deeptailor commented on src/ui/layout/mct-tree.vue:
```diff
@@ -10,107 +13,452 @@
+                        this.updatevisibleItems();
                     });
+                } else {
+                    window.setTimeout(this.setContainerHeight, RECHECK_DELAY);
```
Is there a case where this would need to happen in a timeout?

jvigliotta replied:
Yeah this is a tricky one, I couldn't have this work reliably without the timeout. Just checked again, if I remove it, the height is all janked up on a fresh load or reload of the page. Open to options on this one...

deeptailor commented:
@jvigliotta There are some conflicts with master, once fixed we can merge

deeptailor commented:
@jvigliotta - just noticed that the root items dont have the context menu

jvigliotta commented:
> @jvigliotta - just noticed that the tree items dont have the context menu when navigated to that item.

I'll take at look at this Deep, hopefully it's just a quick fix

deeptailor commented:
## Reviewer Checklist
Changes appear to address issue? Yes
Appropriate unit tests included? Yes
Code style and in-line documentation are appropriate? Yes
Commit messages meet standards? Yes
Has associated issue been labelled unverified? Yes
